styx-api-service/src/test/java/com/spotify/styx/api/WorkflowResourceTest.java

```diff
@@ -816,14 +829,29 @@ public void shouldReturnWorkflows() throws Exception {
     assertJson(response, "[*]", hasSize(2));
   }
 
+  @Test
+  public void shouldReturnFilteredDeploymentTypeWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    when(storage.workflows()).thenReturn(FLYTE_DEPLOYMENT_WORKFLOWS);
```

Comment:
```suggestion
    when(storage.workflows()).thenReturn(buildWorkflowsMap(
      createWorkflowWithDeploymentType("test1", "remote-workflow"),
      createWorkflowWithDeploymentType("test2", "remote-workflow"),
      createWorkflowWithDeploymentType("test3", "")
    ));
```
Having this constant up there make it hard to really grasp what is happening and more importantly if this is correct. Here I am suggesting that you create:
* `createWorkflowWithDeploymentType` for creating a workflow with specified deployment type
* `buildWorkflowsMap` this is a small method that takes an array of workflows and return the map as required
```java
private Map<WorkflowId, Workflow> buildWorkflowsMap(Workflow... workflows) {
   return Arrays.stream(workflows).collect(Collectors.toMap(w -> w.id(), w -> w);
}
```
The `...` are in java called [var args](https://www.baeldung.com/java-varargs)

Then when you do the assertion, you can check that only test1 and test2 are in the response but test3 is not

Reply:
I think the idea is that some workflows have deployment type and some deployment time, right? Just leaving a comment here so we keep in mind that we would need to incorporate `deploymentTime` as an extra parameter if we go this route.

Reply:
I've created a helper class to deal with those. Let me know if that works.